flink-jobs/src/main/java/com/github/projectflink/streaming/ThroughputMeasuringFlatMap.java

```diff
@@ -0,0 +1,72 @@
+package com.github.projectflink.streaming;
+
+import org.apache.flink.api.common.functions.FlatMapFunction;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.util.Collector;
+
+import com.github.projectflink.streaming.Throughput.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+
+class ThroughputMeasuringFlatMap implements FlatMapFunction<Type, Integer> {
+	private static final Logger LOG = LoggerFactory.getLogger(ThroughputMeasuringFlatMap.class);
+
+	private static final long serialVersionUID = -4881110695631095859L;
+
+	private Integer host;
+	private long received;
+	private long start;
+	private int payload;
+	private long logfreq;
+	private long lastLog = -1;
+	private long lastElements;
+
+	public ThroughputMeasuringFlatMap(int payload, long logfreq) {
+		this.payload = payload;
+		this.logfreq = logfreq;
+	}
+
+	@Override
+	public void flatMap(Throughput.Type element, Collector<Integer> collector) throws Exception {
+		if (host == null) {
+			host = Throughput.convertHostnameToInt(InetAddress.getLocalHost().getHostName());
+		}
+		if (start == 0) {
+			start = System.currentTimeMillis();
+		}
+
+		received++;
+		if (received % logfreq == 0) {
+			// throughput over entire time
+			long now = System.currentTimeMillis();
+			long sinceSec = ((now - start) / 1000);
+			if (sinceSec == 0) return;
+			LOG.info("Received {} elements since {}. Elements per second {}, GB received {}",
+					received,
+					sinceSec,
+					received / sinceSec,
+					(received * (8 + 8 + 4 + payload)) / 1024 / 1024 / 1024);
```

Comment:
Hey, this even optimises the log statement by not having to extract the payload from the parameters over and over again - nice! However, the reason for the sum is now outside of this class (defined by `Throughput.Type`) and therefore, it should also be calculated there.